src/load_tilejson.js

```diff
@@ -7,16 +7,40 @@ var request = require('./request'),
 module.exports = {
     _loadTileJSON: function(_) {
         if (typeof _ === 'string') {
-            _ = format_url.tileJSON(_, this.options && this.options.accessToken);
-            request(_, L.bind(function(err, json) {
-                if (err) {
-                    util.log('could not load TileJSON at ' + _);
-                    this.fire('error', {error: err});
-                } else if (json) {
-                    this._setTileJSON(json);
-                    this.fire('ready');
-                }
-            }, this));
+            if (_.indexOf('mapbox://') > -1) {
```

Comment:
This code should live in style_layer.js. This function is called `loadTileJSON` -- it shouldn't gain responsibilities beyond that.